Library/Dependencies/xr/Source/System.cpp

```diff
@@ -0,0 +1,620 @@
+#include <XrPlatform.h>
+#include <XR.h>
+
+#include <assert.h>
+
+namespace
+{
+    struct SupportedExtensions
+    {
+        SupportedExtensions()
+            : Names{}
+        {
+            uint32_t extensionCount{};
+            XR_CHECK(xrEnumerateInstanceExtensionProperties(nullptr, 0, &extensionCount, nullptr));
+            std::vector<XrExtensionProperties> extensionProperties(extensionCount, { XR_TYPE_EXTENSION_PROPERTIES });
+            XR_CHECK(xrEnumerateInstanceExtensionProperties(nullptr, extensionCount, &extensionCount, extensionProperties.data()));
+
+            // D3D11 extension is required for this sample, so check if it's supported.
+            for (const char* extensionName : REQUIRED_EXTENSIONS)
+            {
+                if (!TryEnableExtension(XR_KHR_D3D11_ENABLE_EXTENSION_NAME, extensionProperties))
```

Comment:
This can't be here, no platform-dependent code is allowed in this file. Use the name defined in XrPlatform.h.